### DISCUSSION TITLE
Node and Edge Metadata

### DISCUSSION THREAD

**Discussion by AndrewHannigan** [networkit/networkit#678] [Q&A]
I was wondering if there is any way to associate metadata with a node or an edge.  As a motivating example, suppose I have a large graph of financial transactions and I wish to remove transactions from the graph that don't fall within some date range.  Must this filtering be down prior to the instantiation of the graph, or is there a native way to associate metadata a node/edge?  

And assuming the answer is no (since nodes are represented as ints), does anyone have a recommended approach to keeping track of metadata on nodes/edges for purposes of filtering nodes/edges from the graph?

**Comment by avdgrinten** [comment]
Currently, the answer is no, but support is planned (see #606).

The usual workaround is to store the meta data in an array indexed by node ID (or edge ID).

**Comment by AndrewHannigan** [comment]
@avdgrinten Thank you for the feedback. Any possibility you could point more towards any examples of projects that implement this functionality? Curious how the custom metadata array would be made available back to the python networkit Graph object.

**Comment by MikeB2019x** [comment]
I'd like to pile on to @AndrewHannigan request/comment.  My graphs are usually attribute rich esp. for the edges.  Networkx allows the use of a dict to capture attributes for both nodes and edges.  The answer suggested by @avdgrinten makes me think something similar might be possible here.  If so how is `store the meta data in an array indexed by node ID (or edge ID)` being done?  Any examples?
